utils/backup/S3/fo-backup-s3

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+# Copyright (C) Siemens AG 2017
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# version 2 as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+if [[ $(id -u) -ne 0 ]]; then
+ echo "Please run as root"
+ exit 1
+fi
+
+if [[ ${1} = "-l" ]]; then
+ LOGGING=${2}
+ if [ -z ${LOGGING} ]; then
+  echo "Log dir not set."
+  exit 1
+ fi
+fi
+
+if [ -z ${BUCKET} ]; then
+ echo "Bucket env variable needs to be set"
+ exit 1
+fi
+
+if [ -z ${FILENAME} ]; then
+ FILENAME=${HOSTNAME}.tar
+fi
+
+if [ -n "${ENDPOINT}" ]; then
+  ENDPOINT="--"${ENDPOINT}
+else
+  ENDPOINT=""
+fi
+
+./install
+
+#### check for AWS config file
+if [ ! -f ~/.aws/config ]
+then
+	venv/bin/aws configure
+fi
+
+if [ -n "${LOGGING}" ]; then echo Starting backup at `date` >> ${LOGGING}; fi
+
+#### stop fossology scheduler
+echo "Stopping fossology scheduler..."
+/etc/init.d/fossology stop
+
+#### stop apache2
+echo "Stopping apache2..."
+/etc/init.d/apache2 stop
+
+#### copy database to /srv/fossology
+echo "Backing up database..."
+/etc/init.d/postgresql restart
+
+if [ -n "${LOGGING}" ]; then echo Creating diff of repository at `date` >> ${LOGGING}; fi
+echo "Creating repo diff..."
+ls -RlahGg  --time-style='+' /srv/fossology/repository/ | cut -d ' ' -f 2-5 | venv/bin/aws s3 cp --sse AES256 - s3://${BUCKET}/${FILENAME}-repo-diff.txt
+
+if [ -n "${LOGGING}" ]; then echo Start database dump at `date` >> ${LOGGING}; fi
+#### these steps are split, because of permissions
+su postgres -c 'pg_dumpall' > /tmp/database.txt
+cp /tmp/database.txt /srv/fossology/
+if [ -n "${LOGGING}" ]; then echo Done database dump. Starting Upload at `date` >> ${LOGGING}; fi
+
+#### sync repository
+echo "Uploading and starting backup to S3..."
+tar -zcvf - /srv/fossology/ | venv/bin/aws --debug s3 cp --expected-size 200000000000 --sse AES256 - s3://${BUCKET}/${FILENAME}
+
+if [ -n "${LOGGING}" ]; then echo Done upload. Cleanup at `date` >> ${LOGGING}; fi
+
+#### start apache2
+echo "Starting apache2..."
+/etc/init.d/apache2 start
```

Comment:
You should only start the servers if they were running before starting this script